File: Insertion Sort/Python/insertion_sort.py
list = [5,2,4,6,1,3]

def insertion_sort(sequence,steps=False):
    for j in range(1,len(sequence)):
        item = sequence[j]
        i = j - 1
        # Insert item (element j) in the sorted sequence sequence[0...j-1] 
        while i >= 0 and sequence[i] > item:
            sequence[i + 1] = sequence[i]
            i = i - 1
        sequence[i + 1] = item
        # sorting steps (optional parameter)
        if steps == True:
            print("step ",j,sequence)
    return sequence

print(insertion_sort(list,True))

File: Insertion Sort/C/insertion_sort.c
#include <stdio.h>

int list[] = {5,2,4,6,1,3};
int size = 6;
void insertion_sort(int * list, int size){
    for(int j = 1; j < size; j++){
        int item = list[j];
        int i = j - 1;
        while(i >= 0 && list[i] > item){
            list[i + 1] = list[i];
            i = i - 1;
        }
        list[i+1] = item; 
    }
    return;
}

int main(){

    insertion_sort(list,size);
    for(int i = 0; i < size; i++){
        printf("%d ",list[i]);
    }
    return 0;
}